third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -2840,6 +2879,35 @@ func cidrOrSizeDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 	return strings.HasPrefix(new, "/") && strings.HasSuffix(old, new)
 }
 
+// We want to suppress diffs for the default auto provisioning defaults
+func containerClusterAutoProvisioningDefaultSuppress(k, old, new string, d *schema.ResourceData) bool {
+	o, n := d.GetChange("cluster_autoscaling.0.auto_provisioning_defaults.0.oauth_scopes")
+
+	defaultScopes := []string{
+		"https://www.googleapis.com/auth/logging.write",
+		"https://www.googleapis.com/auth/monitoring",
```

Comment:
Was it your experience that `monitoring` was a default sometimes, and other times it was `monitoring.write`? Did the default change based on `logging_service` and `monitoring_service` values?

Reply:
Good catch, yes, it did change.  When `monitoring_service` was `none` the default was `monitoring.write` and when it is not `none` the default is `monitoring`.

Reply:
Talked in Slack- let's only suppress `monitoring.write`. While monitoring and `logging.write` are conditionally enabled based on GKE Stackdriver Monitoring being enabled, I think it's a valid behaviour for Terraform to show a diff in that case- the user's config isn't the real state.